File: BulkyBookWeb/Models/Product.cs
using BulkyBookWeb.Data.Repository.IRepository;
using Microsoft.AspNetCore.Mvc.ModelBinding.Validation;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BulkyBookWeb.Models
{
    public class Product:IBaseRepository
    {
        [Key]
        public int Id { get; set; }
        [Required]
        public string Title { get; set; }

        [Required]
        public string Description { get; set; }

        [Required]
        public string ISBN { get; set; }

        [Required]
        [Range(1, 100, ErrorMessage = "range between 100")]
        public double ListPrice { get; set; }

        [Required]
        [Range(1, 100, ErrorMessage = "range between 100")]
        public double Price { get; set; }

        [Required]
        [Range(1, 100, ErrorMessage = "range between 100")]
        public double Price50 { get; set; }

        [Required]
        [Range(1, 100, ErrorMessage = "range between 100")]
        public double Price100 { get; set; }

        [ValidateNever]
        public string Image { get; set; }

        //Relationship
        public int CategoryId { get; set; }
        [ForeignKey(nameof(CategoryId))]
        [ValidateNever]
        public Category Category { get; set; }  

        public int CoverTypeId { get; set; }
        [ForeignKey(nameof(CoverTypeId))]
        [ValidateNever]
        public CoverType CoverType { get; set; }

    }
}

File: BulkyBookWeb/wwwroot/js/product.js
var dataTable;
$(document).ready(function () {
    loadDataTable();
});

function loadDataTable() {
    dataTable = $('#myTable').DataTable({
        "ajax": {
            "url":"Product/GetAll"
        },
        "columns": [
            { "data": "image", "width": "15%" },
            { "data": "title", "width": "15%" },
            { "data": "description", "width": "15%" },
            { "data": "isbn", "width": "15%" },
            { "data": "listPrice", "width": "15%" },
            { "data": "price", "width": "15%" },
            { "data": "price50", "width": "15%" },
            { "data": "price100", "width": "15%" },
            { "data": "category.name", "width":"15%" },
            { "data": "coverType.name", "width": "15%" },
            {
                "data": "id",
                "render": (data) => {
                    return `
                        <a href="#" class="btn btn-warning">Edit</a>
                          <a href="#" class="btn btn-danger">Delete</a>
`
                }
            },

            ]
        });
}